ios/RNShopify.m

```diff
@@ -0,0 +1,133 @@
+
+#import "RNShopify.h"
+#import "Buy.h"
+
+@implementation RNShopify {
+  RCTPromiseResolveBlock _resolve;
+  RCTPromiseRejectBlock _reject;
+}
+
+- (dispatch_queue_t)methodQueue
+{
+    return dispatch_get_main_queue();
+}
+RCT_EXPORT_MODULE()
+
+RCT_EXPORT_METHOD(initialize:(NSString *)domain key:(NSString *)key)
+{
+
+  self.client = [[BUYClient alloc] initWithShopDomain:domain
+                                                     apiKey:key
+                                                      appId:@"8"];
+}
+
+RCT_EXPORT_METHOD(getShop:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+
+  [self.client getShop:^(BUYShop *shop, NSError *error) {
+    if (shop && !error) {
+      resolve(shop.domain);
+    }
+    else {
+      reject(@"no_shop", @"There was no shop", error);
+    }
+  }]; 
+}
+
+RCT_EXPORT_METHOD(getProducts:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+
+   [self.client getProductsPage:1 completion:^(NSArray *products, NSUInteger page, BOOL reachedEnd, NSError *error) {
+      if (error == nil && products) {
+        NSMutableArray *result = [NSMutableArray array];
+        for (BUYProduct *product in products) {
+          NSMutableDictionary *dictionary = [product JSONDictionary];
+
+          NSMutableArray *variants = [NSMutableArray array];
+
+          for (BUYProductVariant *variant in product.variants) {
+            NSMutableDictionary *variantDictionary = [variant JSONDictionary];
+
+            NSMutableArray *options = [NSMutableArray array];
+
+            for (BUYOptionValue *option in variant.options) {
+              [options addObject: [option JSONDictionary]]; 
+            }
+
+            variantDictionary[@"options"] = options; 
+
+            [variants addObject: variantDictionary]; 
+          }   
+
+          dictionary[@"variants"] = variants; 
+
+          [result addObject: dictionary];
+        }
+        
+        resolve(result);
+      }
+      else {
+        reject(@"no_products", @"There were no products", error);
+      }
+  }];
+}
+
+RCT_EXPORT_METHOD(checkout:(NSArray *)variants resolver:(RCTPromiseResolveBlock)resolve 
+  rejecter:(RCTPromiseRejectBlock)reject)
+{
+  _resolve = resolve;
+  _reject = reject;
+
+  BUYModelManager *modelManager = self.client.modelManager;
+	BUYCart *cart = [modelManager insertCartWithJSONDictionary:nil];
+
+  for (NSDictionary *dictionary in variants) {
+      BUYProductVariant *variant = [[BUYProductVariant alloc] initWithModelManager:modelManager 
+        JSONDictionary:dictionary];
+     	[cart addVariant:variant];
+  }
+
+	BUYCheckout *checkout = [modelManager checkoutWithCart:cart];
```

Comment:
Fix indentation, is seems that you are using tabs in some lines, and spaces in others.